tmail-backend/mailets/src/main/java/com/linagora/tmail/mailet/OpenAIMailet.java

```diff
@@ -0,0 +1,143 @@
+package com.linagora.tmail.mailet;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import jakarta.mail.Address;
+import jakarta.mail.Message;
+import jakarta.mail.MessagingException;
+import jakarta.mail.internet.InternetAddress;
+import jakarta.mail.internet.MimeMessage;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.MaybeSender;
+import org.apache.james.util.MimeMessageUtil;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetException;
+import org.apache.mailet.base.GenericMailet;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Strings;
+
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+
+/**
+ * OpenAIMailet is a mailet designed to automatically reply to emails
+ * sent to a specific recipient (e.g., gpt@linagora.com). It uses the OpenAI
+ * API to generate responses.
+ *
+ * <p>
+ * Configuration example:
+ *
+ * <pre>
+ * {@code
+ * <mailet match="com.linagora.tmail.mailet.RecipientsContain=gpt@linagora.com" class="com.linagora.tmail.mailet.OpenAIMailet">
+ *     <apiKey>demo</apiKey>
+ *     <gptAddress>gpt@tmail.com</gptAddress>
+ * </mailet>
+ * }
+ * </pre>
+ * </p>
+ *
+ * <p>
+ * The mailet configuration should include the following parameters:
+ * </p>
+ * <ul>
+ * <li><b>apiKey</b>: The API key for accessing the OpenAI service.</li>
+ * <li><b>gptAddress</b>: The email address used to send replies generated by the OpenAI model.</li>
+ * </ul>
+ */
+public class OpenAIMailet extends GenericMailet {
+    public static final String API_KEY_PARAMETER_NAME = "apiKey";
+    public static final String GPT_ADDRESS_PARAMETER_NAME = "gptAddress";
+
+    private ChatLanguageModel openAiModel;
+    private MailAddress gptAddress;
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        if (gptLoopDetected(mail)) {
+            return;
+        }
+
+        try {
+            String prompt = MimeMessageUtil.asString(mail.getMessage());
```

Comment:
Do we still and sanitize HTML?
Is mime multipart (mixed, related, alternative) handled?

Reply:
This method writes the `MimeMessage` to a outputstream.

Result:
```
Received: from localhost (HELO james.org) ([127.0.0.1])
          by hp-quanth (JAMES SMTP Server ) with SMTP ID 9f40f9d1
          for <gpt@james.org>;
          Fri, 19 Jul 2024 09:12:44 +0700 (ICT)
Date: Fri, 19 Jul 2024 09:12:44 +0700 (ICT)
Sender: bob@james.org
To: gpt@james.org
Message-ID: <654233182.0.1721355164402@hp-quanth>
Subject: How can I cook an egg?
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 7bit

I do not know how to cook an egg. Please help me.
```

It seems to be an eml to me. And OpenAI seems smart enough to understand the important part.

Before, I tried to extract only the subject and body, but it seems complicated to extract the body and I am not confident handles well all the cases.
![image](https://github.com/user-attachments/assets/0c546eb3-25fc-43cb-9a37-a53b8195ff3e)

So a bit of no brain here...

Reply:
Discard the subject

How about using `MessageContentExtractor` ?